academy/4-my-own-chain/game-fifo.md

```diff
@@ -69,13 +69,17 @@ When terminating expired games in `EndBlock`, you deal with the expired games th
 
 `k` is still an unbounded number of operations. However, if you use the same expiration duration on each game, for `k` games to expire together in a block they would all have to have had a move in the same previous block (give or take the block before or after). In the worst case, the largest `EndBlock` computation will be proportional to the largest regular block in the past. This is a reasonable risk to take.
 
-**Remember:** this only works if the expiration duration is the same for all games, instead of being a parameter left to a potentially malicious game creator.
+<HighlightBox type="remember">
+
+This only works if the expiration duration is the same for all games, instead of being a parameter left to a potentially malicious game creator.
+
+</HighlightBox>
 
 ## New information
 
-How do you implement a FIFO from which you extract elements at random positions? Choose a doubly-linked list:
+How do you implement FIFO from which you extract elements at random positions? Choose a doubly-linked list:
```

Comment:
I think it still needs the "a". The word FIFO is typically used as a noun.

Reply:
Shouldn't it be "How do you implement a FIFO queue"?

Reply:
Both me and Andrew agree that it should go without an a.

Reply:
If "FIFO" can be understood as a shortening of "FIFO queue", then "a FIFO" would be fine.

Reply:
> If "FIFO" can be understood as a shortening of "FIFO queue", then "a FIFO" would be fine.

That's how I understand it. FIFO is a data structure, it is not the concept of first-in-first-out. Of course, the data structure implements the concept.